api/src/server.rs

```diff
@@ -385,13 +462,28 @@ where
         stream_id: Option<String>,
         _context: &C,
     ) -> Result<InterestsSortKeySortValuePostResponse, ApiError> {
-        self.post_interests_sort_key_sort_value(sort_key, sort_value, controller, stream_id)
-            .await
-            .or_else(|err| {
-                Ok(InterestsSortKeySortValuePostResponse::InternalServerError(
-                    err,
-                ))
-            })
+        let interest = models::Interest {
+            sep: sort_key,
+            model: sort_value,
+            controller,
+            stream_id,
+        };
+        match self.post_interests(interest).await {
```

Comment:
Normally I would say yes, but in this case I think not... this code is slated to be deleted as soon as the dependencies are are updated, but I'm willing to do it if you push 😄.

I considered adding an enum like that `ApiResponse<T> { Success(T), BadRequest(BadRequestResponse), InternalServerError(InternalServerError)}` for all the Server impl returns rather than returning the API response type directly, but I didn't go that far because I wanted to avoid generics. Also considered having methods return `Result<SomeOk, SharedErrorWith400And500>` that we convert, but I kind of want to stabilize the API and then maybe take a pass to normalize things if it seems a bit wonky still.

Reply:
Ah totally cool, thought it might have been an accidental oversight. All good.